src/pdf.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  Widget
+} from '@phosphor/widgets';
+
+import {
+  IRenderMime
+} from '@jupyterlab/rendermime-interfaces';
+
+declare var require: any;
+let PDF = require('pdfjs-dist');
+PDF = require('pdfjs-dist/web/pdf_viewer');
+
+import '../style/index.css';
+import 'pdfjs-dist/web/pdf_viewer.css';
+
+/**
+ * The MIME type for PDF.
+ */
+export
+const MIME_TYPE = 'application/pdf';
+
+export
+const PDF_CLASS = 'pdfViewer';
+
+export
+const PDF_CONTAINER_CLASS = 'jp-PDFJSContainer';
+
+/**
+ * A class for rendering a PDF document.
+ */
+export
+class RenderedPDF extends Widget implements IRenderMime.IRenderer {
+  constructor() {
+    super({ node: Private.createNode() });
+    PDF.PDFJS.disableWorker = true;
+    this._pdfViewer = new PDF.PDFJS.PDFViewer({
+        container: this.node,
+    });
+  }
+
+  /**
+   * Render PDF into this widget's node.
+   */
+  renderModel(model: IRenderMime.IMimeModel): Promise<void> {
```

Comment:
What is does this syntax accomplish? It doesn't seem to be declared as a `const` or `let`, it's not an anonymous function `(model) => {}` or an explicitly declared function using the `function` keyword… so I'm a little thrown off

Reply:
This is a member function of the `RenderedPDF` class.